### PR TITLE
Terraform Part 2

### PR DESCRIPTION
**Pull request by jamesmcclain**
- [x] Replace load balancer with single spot instance
   - ~~Split shared security group~~

I am going to drop splitting the security groups as a goal for this PR (still want to do so eventually, will create an issue to track).  I ran into difficulties which I can describe in detail if there is interest.

### CODE REVIEW

File: terraform/jupyterhub.tf [hunk 1]
```diff
@@ -1,74 +1,12 @@
...
-  role       = "${aws_iam_role.ecs-service.name}"
-  policy_arn = "arn:aws:iam::aws:policy/AmazonEC2ContainerServiceFullAccess"
-}
-
-resource "aws_iam_instance_profile" "ecs-service" {
-  role = "${aws_iam_role.ecs-service.name}"
-}
-
-resource "aws_launch_configuration" "jupyterhub" {
+resource "aws_spot_instance_request" "cheap_worker" {
```
Comment (hectcastro) [inline terraform/jupyterhub.tf]: You can use `spot_price ` of `aws_launch_configuration` to have potentially "self-healing" pool of at least one instance (provided that the `spot_price` is high enough in the available AZs.

File: terraform/jupyterhub.tf [hunk 2]
```diff
@@ -1,74 +1,12 @@
...
+  ami                  = "${var.ecs_ami}"
   iam_instance_profile = "${var.ecs_instance_profile}"
-  image_id             = "${var.ecs_ami}"
   instance_type        = "m3.xlarge"
   key_name             = "${var.key_name}"
-  security_groups      = ["${aws_security_group.security-group.id}"]
+  security_groups      = ["${aws_security_group.security-group.name}"]
```
Comment (hectcastro) [inline terraform/jupyterhub.tf]: I believe that this is supposed to be IDs.

Comment (jamesmcclain) [reply terraform/jupyterhub.tf]: It is actually kind of strange: using the `id` here causes a runtime error, you have to use `name`.  I have noticed a number of these types of inconsistencies.

File: terraform/jupyterhub.tf (hunk 1 above)
Comment (jamesmcclain) [reply terraform/jupyterhub.tf]: I do not currently have a launch configuration for the EMR cluster.  Would creating a variable for the spot price have roughly the same effect that you had in mind?

**Pull request merged**